build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchTestBasePlugin.java

```diff
@@ -199,5 +200,24 @@ public void execute(Task t) {
                 }
             });
         });
+        configureImmutableCollectionsPatch(project);
+    }
+
+    private void configureImmutableCollectionsPatch(Project project) {
+        String configurationName = "immutableCollectionsPatch";
+        var patchConfiguration = project.getConfigurations().register(configurationName, config -> {
+            config.setCanBeConsumed(false);
+        });
+        project.getDependencies().add(configurationName, project.getDependencies().project(
+            Map.of("path", ":test:immutable-collections-patch", "configuration", "patch")));
+        project.getTasks().withType(Test.class).configureEach(test -> {
+            // TODO: don't do this if some flag set, eg from mrjar plugin...
+            test.dependsOn(patchConfiguration);
+            test.doFirst(t -> {
+                test.jvmArgs(
+                    "--patch-module=java.base=" + patchConfiguration.get().getSingleFile() + "/java.base",
+                    "--add-opens=java.base/java.util=ALL-UNNAMED");
+            });
```

Comment:
```suggestion
            test.getInputs().files(patchedFileCollection);
            test.doFirst(t -> {
                ((Test)t).jvmArgs(
                    "--patch-module=java.base=" + patchedFileCollection.getSingleFile() + "/java.base",
                    "--add-opens=java.base/java.util=ALL-UNNAMED");
            });
```

This ensures we keep gradle configuration cache happy. configuration cache compliant means, we cannot reference Configurations and the test task object within the doFirst block.

Ideally we would use `SimpleCommandLineArgumentProvider` here as we do above, but that does currently not support lazy rersolution. 

We wanna change that to support `org.gradle.api.provider.Provider<String>` instead of plain strings but I can take care of that in a later PR unrelated to this one.

Reply:
I'll change as suggestion. One question: how will the configuration be connected to the test task if it's not a dependency?

Reply:
the configuration is marked as input and that is enough for letting gradle do the task wiring for you.